playwright/masonry/scroll-performance.spec.mjs

```diff
@@ -0,0 +1,122 @@
+    .join('\n');
+
+const getFpsFromEvents = (events) => {
+  const fpsTimings = [];
+
+  const drawFrameEvents = events
+    .filter((e) => e.name.includes('DrawFrame'))
+    .sort((a, b) => a.ts - b.ts);
+  const clockSyncEvents = events.filter((e) => e.name.includes('clock_sync'));
+
+  for (let i = 0; i < drawFrameEvents.length - 1; i += 1) {
+    const delta = (drawFrameEvents[i + 1].ts - drawFrameEvents[i].ts) / 10e5;
+    fpsTimings.push({
+      name: 'fps',
+      fps: 1 / delta,
+      ts: drawFrameEvents[i + 1].ts,
+    });
+  }
+
+  return [...fpsTimings, ...clockSyncEvents].sort((a, b) => a.ts - b.ts);
+};
+
+const getFpsAvgPerScroll = (events) => {
+  const fpsAvgPerScroll = [];
+  let fpsSum = 0;
+  let count = 0;
+
+  for (let i = 0; i < events.length; i += 1) {
+    if (events[i].name.includes('clock_sync')) {
+      fpsAvgPerScroll.push(fpsSum / count);
+      fpsSum = 0;
+      count = 0;
+    } else {
+      fpsSum += events[i].fps;
+      count += 1;
+    }
+  }
+
+  return fpsAvgPerScroll;
+};
+
+const createDataCollectedHandler = (events) => (data) => {
+  const event = data.value
+    .filter(
+      (e) => e.name.includes('DrawFrame') || e.name.includes('clock_sync')
+    )
+    .map((e) => ({ name: e.name, ts: e.ts }));
+
+  events.push(...event);
+};
+
+test.describe('Masonry: scrolls', () => {
+  test('scroll test', async ({ page }) => {
+    const performanceTimings = [performance.now()];
+    const events /*: Array<{ name: string, ts: number }> */ = [];
+    let fpsEvents = [];
+    let fpsAvgPerScroll = [];
+
+    const scrollCount = DEFAULT_SCROLL_COUNT;
+
+    const client = await page.context().newCDPSession(page);
+    await client.send('Tracing.start');
+
+    client.on('Tracing.dataCollected', createDataCollectedHandler(events));
+
+    client.on('Tracing.tracingComplete', () => {
+      fpsEvents = getFpsFromEvents(events);
+      fpsAvgPerScroll = getFpsAvgPerScroll(fpsEvents);
+
+      fs.writeFileSync(
+        'playwright/test-results/scroll-performance-timings.csv',
+        eventsToCsv(fpsEvents)
+      );
+    });
+
+    await page.setViewportSize({ width: 1600, height: PAGE_HEIGHT });
+    await page.goto(getServerURL());
+
+    for (let i = 1; i < scrollCount; i += 1) {
+      // await page.waitForTimeout(300);
+      await client.send('Tracing.recordClockSyncMarker', {
+        syncId: 'scrollTimingMark',
+      });
+
+      await page.evaluate(
+        // $FlowFixMe[incompatible-use]
+        ({ pageHeight, index }) => {
+          window.scrollTo({ top: pageHeight * index, behavior: 'smooth' });
+        },
+        { pageHeight: PAGE_HEIGHT, index: i }
+      );
+      await waitForRenderedItems(page, {
+        targetItemsGTE: PINS_COUNT * (i + 1),
+      });
+
+      performanceTimings.push(performance.now());
+    }
+
+    await client.send('Tracing.end');
+
+    // We need this to wait for the tracing data to be completed
+    await page.waitForTimeout(2000);
```

Comment:
I believe you should be able to add a suppression comment here to allow waitForTimeout